server/lib/pbench/__init__.py

```diff
@@ -216,6 +224,18 @@ def __init__(self, cfg_name):
             self.COMMIT_ID = self.conf.get("pbench-server", "commit_id")
         except NoOptionError:
             self.COMMIT_ID = "(unknown)"
+        
+        try:
+            self.logger_type = self.conf.get("logging", "logger")
+        except (NoOptionError, NoSectionError):
+            self.logger_type = "devlog"
+        else:
+            if self.logger_type == "hostport":
+                try:
+                    self.logger_host = self.conf.get("logging", "logger_host")
```

Comment:
AFAICT, there is no definition of `logger_host` in any config file, even if `logger_type` is set to `hostport`.  Yet this exception does not seem to trigger (I don't see any such output in the gold file e.g.) and 26.3 passes. This needs investigating: we should get  (or manufacture) an exception and the test should *NOT* pass.

Reply:
That is, it should not pass if logger_type is set to hostport but logger_host or logger_port (or both) are missing.

Reply:
The upshot is that we need two unit tests for `hostport`: one where we don't define logger_host/logger_port in the (test) config file),  in which case we expect an exception here; and one where they are both defined (logger_host will have to be set to localhost in that case, otherwise a socket() call will fail - DAMHIKT), in which case, the pytest should do its thing.